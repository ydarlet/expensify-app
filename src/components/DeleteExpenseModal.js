import React from 'react';
import { connect } from 'react-redux';
import { openModal, closeModal } from '../actions/modal';
import Modal from 'react-modal';

export class DeleteExpenseModal extends React.Component {
    render () {
        return (
            <Modal
                isOpen={this.props.modal === true}
                onRequestClose={this.props.closeModal}
                contentLabel="Delete Expense"
                closeTimeoutMS={200}
                className="modal"
                shouldCloseOnEsc={true}
                shouldReturnFocusAfterClose={true}
            >
                <div className="box-layout box-layout--no-background">
                    <div className="box-layout__box box-layout__box--dark-blue box-layout__box--delete-expense">
                        <div>
                            <h3 className="modal-title">Are you sure ?</h3>
                            <p>You will not be able to recover this expense.</p>
                            <button 
                                className="button button--secondary button--margin-right" 
                                onClick={this.props.closeModal}
                            >
                                Cancel
                            </button>
                            <button 
                                className="button button--confirm"
                                onClick={this.props.onRemoveExpense}
                            >
                                Yes, delete it!
                            </button>
                        </div>
                    </div>
                </div>

            </Modal>
        );
    };
};

const mapStateToProps = (state, props) => ({
    modal: state.modal
});

const mapDispatchToProps = (dispatch, props) => ({
    openModal: () => dispatch(openModal(true)),
    closeModal: () => dispatch(closeModal(false))
})

export default connect(mapStateToProps, mapDispatchToProps)(DeleteExpenseModal);