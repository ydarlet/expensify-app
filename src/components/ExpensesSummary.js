import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import expensesNbr from '../selectors/expenses-nbr';

export const ExpensesSummary = ({ expenseCount, expensesTotal, expensesNbr }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const expensesNotShowingNbr = expensesNbr - expenseCount;
    const expensesNotShowingWord = expensesNotShowingNbr === 1 ? 'expense' : 'expenses';
    const expensesNotShowingSentence = `Not showing ${expensesNotShowingNbr} ${expensesNotShowingWord} because of filters.`;
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 
                    className="page-header__title">Viewing 
                    <span> {expenseCount}</span> {expenseWord} totalling 
                    <span> {formatedExpensesTotal}</span>
                </h1>
                <h5 className="page-header__sub-title">
                {
                    expensesNotShowingNbr == 0 ? (
                        'Showing all expenses, no filters in place.'
                    ) : expensesNotShowingSentence
                }
                </h5>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div> 
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        expensesNbr: expensesNbr(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);