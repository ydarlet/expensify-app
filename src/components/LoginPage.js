import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginFacebook } from '../actions/auth';

export const LoginPage = ({ startLoginGoogle, startLoginFacebook }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className="button button--google" onClick={startLoginGoogle}>
                <i className="fab fa-google font-awesome"></i>
                Login with Google
            </button>
            <button className="button button--facebook" onClick={startLoginFacebook}>
                <i className="fab fa-facebook-f font-awesome"></i>
                Login with Facebook
            </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginFacebook: () => dispatch(startLoginFacebook())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);