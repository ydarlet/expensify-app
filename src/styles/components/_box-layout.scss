.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout--no-background {
    background: none;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
}

.box-layout__box--delete-expense {
    width: 30rem;
}

.box-layout__box--dark-blue {
    background: $dark-blue;
    color: white;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}