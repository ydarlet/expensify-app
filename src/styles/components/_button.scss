.button {
    background-color: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--confirm {
    background: $confirm-red;
}

.button--facebook {
    background: $facebook-blue2;
    width: 17vw;
}

.button--google {
    background: $google-red2;
    margin-bottom: $s-size;
    width: 17vw;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}

.button--margin-right {
    margin-right: $m-size;
}