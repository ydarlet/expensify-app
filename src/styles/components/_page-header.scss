.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page_header__actions {
    margin-top: $m-size;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
}

.page-header__sub-title {
    font-weight: 300;
    margin: 0 0 $s-size 0.5rem;
}