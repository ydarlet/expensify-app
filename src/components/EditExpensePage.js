import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import { openModal, closeModal } from '../actions/modal';
import DeleteExpenseModal from './DeleteExpenseModal';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemoveExpense = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.closeModal();
        this.props.history.push('/');
    };
    render () {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button
                        className="button button--secondary"
                        onClick={this.props.openModal}
                    >
                        Remove Expense
                    </button>
                    <DeleteExpenseModal 
                        onRemoveExpense={this.onRemoveExpense}
                    />
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    modal: state.modal
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
    openModal: () => dispatch(openModal(true)),
    closeModal: () => dispatch(closeModal(false))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);