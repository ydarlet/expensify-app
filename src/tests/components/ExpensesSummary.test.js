import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={23512340987} />);
    expect(wrapper).toMatchSnapshot();
});
