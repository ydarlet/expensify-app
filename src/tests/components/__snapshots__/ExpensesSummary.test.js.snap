// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`Should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $235,123,409.87
  </h1>
</div>
`;
